rasa/core/actions/forms.py

```diff
@@ -679,9 +700,10 @@ async def do(
     ) -> List[Event]:
         events = await self._validate_if_required(tracker, domain, output_channel, nlg)
 
-        events += await self.request_next_slot(
-            tracker, domain, output_channel, nlg, events_so_far + events
-        )
+        if not await self.user_rejected_manually(events):
```

Comment:
could you please add a test for this logic?

Reply:
This logic is already tested by this test, which I had to modify after the update, because the next slot is no longer being requested when the user manually rejects:

https://github.com/RasaHQ/rasa/blob/fc674407543df7ca120d3115f5ce46f27df75ada/tests/core/actions/test_forms.py#L295-L302